test/e2e/upgrade/manifestdelete/manifest-delete.go

```diff
@@ -0,0 +1,192 @@
+package manifestdelete
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	restclient "k8s.io/client-go/rest"
+	apiregistrationsclientset "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/upgrades"
+)
+
+// UpgradeTest contains artifacts used during test.
+type UpgradeTest struct {
+	oc     *exutil.CLI
+	config *restclient.Config
+}
+
+type object struct {
+	kind      string
+	namespace string
+	name      string
+}
+
+type unstructured struct {
+	name      string
+	namespace string
+	group     string
+	version   string
+	resource  string
+}
+
+var (
+	// Update this array to contain objects which should have been deleted in this release. For example:
+	//	object{kind: "Namespace",
+	//		namespace: "",
+	//		name:      "openshift-foobar"},
+	//
+	deletes = []object{}
+
+	// Update this array to contain unstructured objects which should have been deleted in this release. For example:
+	//	unstructured{name: "dns",
+	//		namespace: "openshift-dns-operator",
+	//		group:     "monitoring.coreos.com",
+	//		version:   "v1",
+	//		resource:  "prometheusrules"},
+	//
+	unstructuredDeletes = []unstructured{{
+		name:      "kube-apiserver",
+		namespace: "openshift-kube-apiserver",
+		group:     "monitoring.coreos.com",
+		version:   "v1",
+		resource:  "prometheusrules",
+	}}
+)
+
+func (obj object) uniqueName() string {
+	if len(obj.namespace) == 0 {
+		return obj.name
+	}
+	return obj.namespace + "/" + obj.name
+}
+
+func (obj object) String() string {
+	return fmt.Sprintf("%s %s", obj.kind, obj.uniqueName())
+}
+
+func (u unstructured) uniqueName() string {
+	if len(u.namespace) == 0 {
+		return u.name
+	}
+	return u.namespace + "/" + u.name
+}
+
+func (u unstructured) String() string {
+	return fmt.Sprintf("unstructured %s/%s, %s, %s", u.group, u.version, u.resource, u.uniqueName())
+}
+
+func (UpgradeTest) Name() string { return "check-for-deletes" }
+func (UpgradeTest) DisplayName() string {
+	return "[sig-arch] Verify object deletions after upgrade success"
```

Comment:
This is related to the CVO behavior, not to sig-arch.  `[bz-Cluster Version Operator]` would match what we did for operators.

Reply:
changed